### PR TITLE
Add FCPath.name

### PR DESCRIPTION
**Pull request by rfrenchseti**
# Fixed Issues

N/A

# Summary of Changes

- Added `FCPath.name` method.
- Fixed minor typos in README.

# Known Problems

N/A
 
 <div id='description'>
<h3>Summary by Bito</h3>
This PR enhances the FCPath class by adding a 'name' property to return the final path component. It improves FCPath initialization to handle FCPath instances and updates the test suite. The changes include modifying existing methods and removing some test functions, likely to be replaced with more specific tests.
<br>
<br>
<b>Unit tests added</b>: True
<br>
<br>
<b>Estimated effort to review (1-5, lower is better)</b>: 1
</div>

### CODE REVIEW

**Comment by pds-admin** [comment]
AI Code Review is in progress (usually takes <3 minutes unless it's a very large PR).

**Comment by pds-admin** [comment]
<div>
<h2>Changelist by <a href="https://bito.ai">Bito</a></h2>
<p>This pull request implements the following key changes.</p>

<div>
<table width="100%">
<tbody>
  <tr>
    <th>Key Change</th>
    <th>Files Impacted</th>
  </tr>

<tr>
<td width="30%">
    <b>Feature Improvement</b> - Enhanced FCPath Functionality
</td>
<td width="70%">


<div>

-&nbsp;[`file_cache_path.py`](https://github.com/SETI/rms-filecache/pull/26/files#diff-cf8cb5ca46759fabc5de6f55c831cc22e9c1784b7ac55e9f5fe7be647907d9c6) - Added FCPath.name property and improved FCPath initialization
<div>

-&nbsp;[`test_file_cache_path.py`](https://github.com/SETI/rms-filecache/pull/26/files#diff-5051c3309a20e9959f154e32933c6fbc2083309540f28dc3981bb0e317356a4f) - Updated tests for FCPath, including new tests for the name property
</div>
</td>
</tr>

</tbody>
</table>
</div>
</div>

File: filecache/file_cache_path.py [hunk 1]
```diff
@@ -473,6 +476,12 @@
         else:
             return FCPath(other, self._path, copy_from=self)
 
+    @property
+    def name(self) -> str:
+        """The final component of the path."""
+
+        return FCPath._split(self._path)[1]
```
Comment (pds-admin) [inline filecache/file_cache_path.py:483]: <div>


<div id="suggestion">
<div id="issue"><b>Consider optimizing path splitting operation</b></div>
<div id="fix">

Consider using `self._split_parts()` directly instead of `_split()` since we only need the final component. The `_split()` method does additional processing that isn't needed here.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -483 +483 @@
 -        return FCPath._split(self._path)[1]
 +        _, _, subpath = FCPath._split_parts(self._path)
 +        return subpath.split('/')[-1] if subpath else ''
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/SETI/rms-filecache/pull/26#issuecomment-2499489551>#aa998b</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

File: tests/test_file_cache_path.py [hunk 2]
```diff
@@ -434,6 +412,23 @@ def test_rtruediv():
...
+    p = FCPath('c.txt')
+    assert p.name == 'c.txt'
+    p = FCPath('/c.txt')
+    assert p.name == 'c.txt'
+    p = FCPath('a/b/c.txt')
+    assert p.name == 'c.txt'
+    p = FCPath('C:/a/b/c.txt')
+    assert p.name == 'c.txt'
+    p = FCPath('http://bucket/a/b/c')
+    assert p.name == 'c'
```
Comment (pds-admin) [inline tests/test_file_cache_path.py:429]: <div>


<div id="suggestion">
<div id="issue"><b>Consider adding more edge case tests</b></div>
<div id="fix">

Consider adding test cases for edge cases such as paths with multiple dots or paths ending with a directory separator. This could help ensure the `name` property handles all scenarios correctly.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -429,2 +429,8 @@ def test_name():
      assert p.name == 'c'
 +    p = FCPath('file.with.multiple.dots.txt')
 +    assert p.name == 'file.with.multiple.dots.txt'
 +    p = FCPath('path/with/trailing/slash/')
 +    assert p.name == ''
 +    p = FCPath('C:\\Windows\\path\\')
 +    assert p.name == 'path'
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/SETI/rms-filecache/pull/26#issuecomment-2499489551>#aa998b</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

**Comment by pds-admin** [comment]
## Code Review Agent Run Status

- **Limitations and other issues:** <small>❌</small>&nbsp;Failure - We encountered technical difficulties while attempting to generate code feedback. Please try again or contact support@bito.ai.
